### ISSUE TITLE
general comments

### ISSUE THREAD

**Issue by kpurdon** [enkhalifapro/go-fast#1]
A few general comments:
- not sure presenting gin over net/http to beginning users is the right way to go.
- filenames are generally some_thing.go not someThing.go
- package names are generally something not someThing
- consider reading [this](https://medium.com/@PurdonKyle/building-a-unit-testable-api-in-golang-b42ff1fcbae7#.etfru1bx5) for a "better" way to set up things like this for testing
- if this is really for new users to go you should have a much more thorough README describing what it does, what libs your using, why it's laid out how it is, how to extend it, ...
- I would also expect many more information comments in code for a "starter template"

Hope these are helpful comments.

**Comment by enkhalifapro** [comment]
- not sure presenting gin over net/http to beginning users is the right way to go.
  - yes we built starter template based on gin because it's so performant and easy to use
- filenames are generally some_thing.go not someThing.go
  - Done
- package names are generally something not someThing
  - Done
- consider reading [this](https://medium.com/@PurdonKyle/building-a-unit-testable-api-in-golang-b42ff1fcbae7#.etfru1bx5) for a "better" way to set up things like this for testing
- Done
- if this is really for new users to go you should have a much more thorough README describing what it does, what libs your using, why it's laid out how it is, how to extend it, ...
  - it's still under construction
- I would also expect many more information comments in code for a "starter template"
  - in consideration

**Comment by kpurdon** [comment]
> yes we built starter template based on gin because it's so performant and easy to use

Just an FYI, so is the standard net/http and it will be around much longer than Gin. =)

**Comment by enkhalifapro** [comment]
yeah @kpurdon  but gin is built on top of [httprouter](https://github.com/julienschmidt/httprouter) which is very faster than  net/http

**Comment by kpurdon** [comment]
Correct, but you can use httprouter standalone (with net/http) and get rid of all the extra gin stuff you don't need (which is what I, and a lot of others do). I won't really get into the framework v non-framework but will say that you should explicitly call out in the README that this is a starter template for Gin. Even though the majority of the community would recommend starting with net/http and pulling in libs (like httprouter,...).

I still think this is fine work, just am actively trying to avoid teaching new-to-golang people a "framework first" approach which is not the "idiomatic" way to start with Golang.

**Comment by enkhalifapro** [comment]
ok @kpurdon can you contribute with us and write the best for this project

**Comment by kpurdon** [comment]
What I would write would be something very different than what's currently here so that might not be very helpful.

**Comment by enkhalifapro** [comment]
@kpurdon your contribution is welcome whatever it is

**Comment by enkhalifapro** [comment]
let's share knowledge and build something useful for new gophers

**Comment by kpurdon** [comment]
I'll think on this a bit and get back to you. https://github.com/kpurdon/go-api-example is something I wrote for the article I linked. There are also a lot of good resources out there for getting started with net/http so i'm not 100% sure I want to add more to that space. Also i'm not super interested in building templates right now (more into building real apps).

I'll gladly lend code review if you want to keep working here.
